src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -314,12 +314,15 @@ void instrument_spec_assignst::instrument_instructions(
   goto_programt::targett instruction_it,
   const goto_programt::targett &instruction_end,
   skip_function_paramst skip_function_params,
-  optionalt<cfg_infot> &cfg_info_opt)
+  optionalt<cfg_infot> &cfg_info_opt,
+  const std::function<bool(const goto_programt::targett &)> &filter)
 {
   while(instruction_it != instruction_end)
   {
     // Skip instructions marked as disabled for assigns clause checking
-    if(has_disable_assigns_check_pragma(instruction_it))
+    if(
+      has_disable_assigns_check_pragma(instruction_it) ||
+      (filter && !filter(instruction_it)))
```

Comment:
See below: I'm not convinced we should permit an empty `std::function` here.

Reply:
Or at least we should make it the default to check all instructions.

Reply:
That _is_ the default.